### PR TITLE
[doc] Add shape for table widget

### PR DESCRIPTION
**Pull request by lfasani**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/3823

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [X] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: doc/iterations/2024.11/add_table_widget.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+The table widget is made of lines and columns.
+It also contains a text area to filter the lines.
+A `Line` is associated with a semantic element.
+A `Cell`, which is the intersection between a line and a column, is also associtaed with a semantic element.
+
+
+In this table, it is possible to:
+
+* *Order*: When clicking/clicking again on the column header, order the lines based on alphanumerical/inverse alphanumerical value.
+* *Filter*: When entering a string in the filter area, to filter lines. A line is displayed if any of its cell contains the filtering string be it alpha or numeric.
```
Comment (frouene) [inline doc/iterations/2024.11/add_table_widget.adoc]: one sentence per line

File: doc/iterations/2024.11/add_table_widget.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* *Resize* columns(not persisted)
+* *Export* the content in a csv format
+* *Select*: When selecting a line or a cell, the selection is changed with the semantic element associated with the line or the cell.
+* *Edit a cell*: When clicking inside the cell, it is possible to change the content. 
+The change is sent to the server when focussing out.
+** drop down list for mono-valued feature
+** list when check button for multi-valued feature
```
Comment (frouene) [inline doc/iterations/2024.11/add_table_widget.adoc]: list WITH check button?

Comment (frouene) [inline doc/iterations/2024.11/add_table_widget.adoc]: How these multi-valued features will be display in the table.
Would it be complicated to use the ref widget?

**Review by frouene — DISMISSED** [review]

File: doc/iterations/2024.11/add_table_widget.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+== Solution
+
+Sirius-Web provide the API so that the developper is able to provide the `FormDescription` containing the `TableWidget`.
+
+The table widget is part of a form representation.
+
+The table widget is made of lines and columns.
+It also contains a text area to filter the lines.
+A `Line` is associated with a semantic element.
+A `Cell`, which is the intersection between a line and a column, is also associtaed with a semantic element.
```
Comment (sbegaudeau) [inline doc/iterations/2024.11/add_table_widget.adoc]: associtaed

File: doc/iterations/2024.11/add_table_widget.adoc (hunk 2 above)
Comment (sbegaudeau) [reply doc/iterations/2024.11/add_table_widget.adoc]: @frouene The goal is to reintroduce our previous prototype as soon as possible and then to move to an updated implementation so this file describe pretty much was we had before.

**Review by sbegaudeau — COMMENTED** [review]
When the release 2024.9.0 will be made, the CHANGELOG for 2024.11.0 will have to be updated to reference this shape

File: doc/iterations/2024.11/add_table_widget.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+== Solution
+
+Sirius-Web provide the API so that the developper is able to provide the `FormDescription` containing the `TableWidget`.
+
+The table widget is part of a form representation.
+
+The table widget is made of lines and columns.
+It also contains a text area to filter the lines.
+A `Line` is associated with a semantic element.
```
Comment (sbegaudeau) [inline doc/iterations/2024.11/add_table_widget.adoc]: column should be selectable too to consider matrixes for example.

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**